scripts/globals/bluemagic.lua

```diff
@@ -285,6 +285,10 @@ xi.spells.blue.usePhysicalSpell = function(caster, target, spell, params)
         hitsdone = hitsdone + 1
     end
 
+    if finaldmg <= 0 then
```

Comment:
Too simple, stoneskin and phalanx can make stuff do 0 dmg without "no effect"

Reply:
Stoneskin and Phalanx modifiers are applied after this by applySpellDamage so if the value calculated here is 0 it will be sent to applySpellDamage as 0. When the spell successfully hits and does damage it will display a 0 damage message correctly since the spell damage was calculated higher than 0 but applySpellDamage then makes the Phalanx/Stoneskin adjustments.

I tested on a mob with Stoneskin and it did properly display both the no effect message as well as the 0 damage message. Please let me know if there's something I might be missing.

Reply:
great, that's all I needed to know